### PR TITLE
feature/mx-1509 update container build and push

### PR DESCRIPTION
**Pull request by cutoffthetop**
# PR Context
- this is a prep for mx-1509 that fixes the container building and pushing to [ghcr](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry)
- this was completely broken for [mex-backend](https://github.com/robert-koch-institut/mex-backend/pkgs/container/mex-backend), and on [mex-drop](https://github.com/robert-koch-institut/mex-drop/pkgs/container/mex-drop) it was missing a `latest` tag
- demo PR for the release pipeline changes: https://github.com/robert-koch-institut/mex-backend/pull/84
- demo container index https://github.com/cutoffthetop/mex-backend/pkgs/container/mex-backend
- demo release page https://github.com/cutoffthetop/mex-backend/releases

# Added
- Add more metadata labels to docker file
- Add hint to PR template to explain testing for template changes
- Configure sensible concurrency rules for all workflows
- Add distribution upload step to release pipeline

# Changes
- Trigger documentation on release pipeline instead of pushed tag
- Harmonize cron schedules

# Removed
- Remove docker workflow and move into release pipeline
- Remove stale dependabot cve-scan config

# Fixed
- Fix tagging and pushing docker containers to ghcr.io

### CODE REVIEW

File: mex-{{ cookiecutter.project_name }}/.github/workflows/release.yml [hunk 1]
```diff
@@ -74,7 +77,57 @@ jobs:
...
+
+  containerize:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    needs: release
+    steps:
+      - name: Build, tag and push docker image to ghcr
+        uses: GlueOps/github-actions-build-push-containers@v0.4.2
+        with:
+          tags: "${{ github.sha }},${{ needs.release.outputs.tag }},latest"
```
Comment (rababerladuseladim) [inline mex-{{ cookiecutter.project_name }}/.github/workflows/release.yml:95]: Escape with raw

File: mex-{{ cookiecutter.project_name }}/.github/workflows/release.yml [hunk 2]
```diff
@@ -74,7 +77,57 @@ jobs:
...
+        with:
+          fetch-depth: 1
+
+      - name: Cache requirements
+        uses: actions/cache@v4
+        env:
+          cache-name: cache-requirements
+        with:
+          path: ~/.cache/pip
+          key: ${{ env.cache-name }}-${{ hashFiles('requirements.txt') }}
```
Comment (rababerladuseladim) [inline mex-{{ cookiecutter.project_name }}/.github/workflows/release.yml:113]: Escape with raw

**Review by rababerladuseladim — CHANGES_REQUESTED** [review]
Nice, only tiny remarks

File: mex-{{ cookiecutter.project_name }}/.github/workflows/release.yml (hunk 2 above)
Comment (cutoffthetop) [reply mex-{{ cookiecutter.project_name }}/.github/workflows/release.yml]: a23701a

File: mex-{{ cookiecutter.project_name }}/.github/workflows/release.yml (hunk 1 above)
Comment (cutoffthetop) [reply mex-{{ cookiecutter.project_name }}/.github/workflows/release.yml]: a23701a

**Review by rababerladuseladim — APPROVED** [review]

**Pull request merged**